public/js/overview.js

```diff
@@ -502,6 +504,54 @@ $(document).ready(function () {
             sessionStorage.setItem('visited', 'true');
         }
     }
+
+    // checks if page button is selected. if not make ajax request to fetch more passwords
+    $('body').on('click', '.page_selector', function() {
+        const pageSelectedNumber = $(this).attr('data-page_number');
+        const pageSeletorButton = $(this);
+
+        if (pageSeletorButton.hasClass('selected')) {
+            return;
```

Comment:
Keep the code simple by checking the opposite and then doing the ajax call inside of the if block.

Reply:
Is it really simpler? Here you see when the condition that cancels further execution

Reply:
In my opinion it is, because you directly see if the button has no class selected it does nothing. (similar to php's `isset` check)